packages/flutter/lib/src/material/carousel.dart

```diff
@@ -490,6 +627,506 @@ class _RenderSliverFixedExtentCarousel extends RenderSliverFixedExtentBoxAdaptor
+      index = round;
+    } else {
+      index = actual.floor();
+    }
+    return constraints.scrollOffset - index * firstChildExtent;
+  }
+
+  // Given the off-screen extent for the first visible item, we can know the
+  // on-screen extent for the first visible item.
+  double get _distanceToLeadingEdge => firstChildExtent - _firstVisibleItemOffscreenExtent;
+
+  // Given an index, this method returns the layout offset for the item. The `index`
+  // is firstly compared to `_firstVisibleItemIndex` and compute the distance
+  // between them, then compute all the current extents for items that are located
+  // in front.
+  @override
+  double indexToLayoutOffset(
+    @Deprecated(
+      'The itemExtent is already available within the scope of this function. '
+      'This feature was deprecated after v3.20.0-7.0.pre.'
+    )
+    double itemExtent,
+    int index,
+  ) {
+    if (index == _firstVisibleItemIndex) {
+      if (_distanceToLeadingEdge <= effectiveShrinkExtent) {
+        return constraints.scrollOffset - effectiveShrinkExtent + _distanceToLeadingEdge;
+      }
+      return constraints.scrollOffset;
+    }
+    double visibleItemsTotalExtent = _distanceToLeadingEdge;
+    for (int i = _firstVisibleItemIndex + 1; i < index; i++) {
+      visibleItemsTotalExtent += _buildItemExtent(i, _currentLayoutDimensions);
+    }
+    return constraints.scrollOffset + visibleItemsTotalExtent;
+  }
+
+  @override
+  int getMinChildIndexForScrollOffset(
+    double scrollOffset,
+    @Deprecated(
+      'The itemExtent is already available within the scope of this function. '
+      'This feature was deprecated after v3.20.0-7.0.pre.'
+    )
+    double itemExtent,
+  ) {
+    return math.max(_firstVisibleItemIndex, 0);
+  }
+
+  @override
+  int getMaxChildIndexForScrollOffset(
+    double scrollOffset,
+    @Deprecated(
+      'The itemExtent is already available within the scope of this function. '
+      'This feature was deprecated after v3.20.0-7.0.pre.'
+    )
+    double itemExtent,
+  ) {
+    final int? childCount = childManager.estimatedChildCount;
+    if (childCount != null) {
+      double visibleItemsTotalExtent = _distanceToLeadingEdge;
+      for (int i = _firstVisibleItemIndex + 1; i < childCount; i++) {
+        visibleItemsTotalExtent += _buildItemExtent(i, _currentLayoutDimensions);
+        if (visibleItemsTotalExtent >= constraints.viewportMainAxisExtent) {
+          return i;
+        }
+      }
+    }
+    return childCount ?? 0;
+  }
+
+  @override
+  double computeMaxScrollOffset(
+    SliverConstraints constraints,
+    @Deprecated(
+      'The itemExtent is already available within the scope of this function. '
+      'This feature was deprecated after v3.20.0-7.0.pre.'
+    )
+    double itemExtent,
+  ) {
+    return childManager.childCount * maxChildExtent;
+  }
+
+  BoxConstraints _getChildConstraints(int index) {
+    double extent;
+    extent = itemExtentBuilder!(index, _currentLayoutDimensions)!;
```

Comment:
Could be simplified:

```suggestion
    final double extent = itemExtentBuilder!(index, _currentLayoutDimensions)!;
```